packages/core/src/auth/tokens/token-scope.ts

```diff
@@ -1,26 +1,66 @@
 export enum CacheRole {
   None = 'none',
   ReadWrite = 'readwrite',
+  ReadOnly = 'readonly',
 }
 
+export class All {}
+export class CacheName {
+  name: string;
+  constructor(name: string) {
+    this.name = name;
+  }
+}
+export type CacheResource = All | CacheName;
```

Comment:
👍 to how you are doing the typing here where you are unioning the things together.

As for the name "Resource" here, I don't hate it, might have a slight pref for something like "Selector" or "Descriptor", but I don't love those either, and would be fine being outvoted

Reply:
I renamed Resource to Selector in the client-protos and am working on doing that here as well!